PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsDotNetCoreIsolatedDebugProfile.kt

```diff
@@ -0,0 +1,169 @@
+import java.time.Duration
+
+class AzureFunctionsDotNetCoreIsolatedDebugProfile(
+        private val dotNetExecutable: DotNetExecutable,
+        executionEnvironment: ExecutionEnvironment)
+    : DebugProfileStateBase(executionEnvironment) {
+
+    companion object {
+        private val logger = Logger.getInstance(AzureFunctionsDotNetCoreIsolatedDebugProfile::class.java)
+        private const val DOTNET_ISOLATED_DEBUG_ARGUMENT = "--dotnet-isolated-debug"
+    }
+
+    private var processId = 0
+    private lateinit var targetProcessHandler: ProcessHandler
+    private lateinit var console: ConsoleView
+
+    override suspend fun createWorkerRunInfo(lifetime: Lifetime, helper: DebuggerHelperHost, port: Int): WorkerRunInfo {
+        // Launch Azure Functions host process
+        launchAzureFunctionsHost()
+
+        // Wait until we get a process ID (or the process terminates)
+        pumpMessages(Duration.ofMinutes(5)) {
+            processId != 0 || targetProcessHandler.isProcessTerminated
+        }
+        if (processId == 0) {
+            logger.error("Azure Functions host did not return isolated worker process id.")
+
+            // REVIEW: If we do not get pid from the isolated worker process, should we destroy the process here?
+            if (!targetProcessHandler.isProcessTerminating && !targetProcessHandler.isProcessTerminated) {
+                targetProcessHandler.destroyProcess()
+            }
+
+            // REVIEW: Should we throw here instead?
+            PluginUtil.showErrorNotification(
+                    RiderAzureBundle.message("run_config.run_function_app.debug.notification.title"),
+                    RiderAzureBundle.message("run_config.run_function_app.debug.notification.isolated_worker_pid_unspecified"))
+        }
+
+        // Create debugger worker info
+        return createWorkerRunInfoFor(port, DebuggerWorkerPlatform.AnyCpu)
+    }
+
+    private fun launchAzureFunctionsHost() {
+        val useExternalConsole = dotNetExecutable.useExternalConsole
+
+        val programParameters = ParametersListUtil.parse(dotNetExecutable.programParameterString)
+        if (!programParameters.contains(DOTNET_ISOLATED_DEBUG_ARGUMENT)) {
+            programParameters.add(DOTNET_ISOLATED_DEBUG_ARGUMENT)
+        }
+
+        val commandLine = createEmptyConsoleCommandLine(useExternalConsole)
+                .withExePath(dotNetExecutable.exePath)
+                .withWorkDirectory(dotNetExecutable.workingDirectory)
+                .withEnvironment(dotNetExecutable.environmentVariables)
+                .withRawParameters(ParametersListUtil.join(programParameters))
+
+        targetProcessHandler = if (useExternalConsole) {
+            ExternalConsoleMediator.createProcessHandler(commandLine)
+        } else {
+            TerminalProcessHandler(commandLine)
+        }
+
+        val commandLineString = commandLine.commandLineString
+
+        logger.info("Functions host process started: $commandLineString")
```

Comment:
`ProcessAdapter` has override for fun `startNotified(event: ProcessEvent)`. Maybe move the "started message" under this event? I would like to log a `commandLine` here though.